instrumentation/reactor-netty-0.9/src/main/java/io/opentelemetry/javaagent/instrumentation/reactor_netty/ReactorHttpClientInstrumentation.java

```diff
@@ -0,0 +1,102 @@
+import static net.bytebuddy.matcher.ElementMatchers.isStatic;
+import static net.bytebuddy.matcher.ElementMatchers.named;
+
+import com.google.auto.service.AutoService;
+import io.netty.bootstrap.Bootstrap;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.javaagent.instrumentation.api.CallDepthThreadLocalMap;
+import io.opentelemetry.javaagent.instrumentation.netty.v4_1.AttributeKeys;
+import io.opentelemetry.javaagent.tooling.Instrumenter;
+import java.util.Map;
+import java.util.function.BiConsumer;
+import java.util.function.BiFunction;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import reactor.core.publisher.Mono;
+import reactor.netty.Connection;
+import reactor.netty.http.client.HttpClient;
+import reactor.netty.http.client.HttpClientRequest;
+
+@AutoService(Instrumenter.class)
+public final class ReactorHttpClientInstrumentation extends Instrumenter.Default {
+
+  public ReactorHttpClientInstrumentation() {
+    super("reactor-httpclient");
+  }
+
+  @Override
+  public ElementMatcher<TypeDescription> typeMatcher() {
+    return named("reactor.netty.http.client.HttpClient");
+  }
+
+  @Override
+  public Map<? extends ElementMatcher<? super MethodDescription>, String> transformers() {
+    return singletonMap(
+        isStatic().and(named("create")),
+        ReactorHttpClientInstrumentation.class.getName() + "$CreateAdvice");
+  }
+
+  @Override
+  public String[] helperClassNames() {
+    return new String[] {
+      ReactorHttpClientInstrumentation.class.getName() + "$MapConnect",
+      ReactorHttpClientInstrumentation.class.getName() + "$OnRequest",
+      "io.opentelemetry.javaagent.instrumentation.netty.v4_1.AttributeKeys",
+      "io.opentelemetry.javaagent.instrumentation.netty.v4_1.AttributeKeys$1",
+      // these below a transitive dependencies of AttributeKeys from above
+      "io.opentelemetry.javaagent.instrumentation.netty.v4_1.client.NettyHttpClientTracer",
+      "io.opentelemetry.javaagent.instrumentation.netty.v4_1.client.NettyResponseInjectAdapter",
+      "io.opentelemetry.javaagent.instrumentation.netty.v4_1.client.HttpClientRequestTracingHandler",
+      "io.opentelemetry.javaagent.instrumentation.netty.v4_1.client.HttpClientResponseTracingHandler",
+      "io.opentelemetry.javaagent.instrumentation.netty.v4_1.client.HttpClientTracingHandler",
+      "io.opentelemetry.javaagent.instrumentation.netty.v4_1.server.NettyHttpServerTracer",
+      "io.opentelemetry.javaagent.instrumentation.netty.v4_1.server.NettyRequestExtractAdapter",
+      "io.opentelemetry.javaagent.instrumentation.netty.v4_1.server.HttpServerRequestTracingHandler",
+      "io.opentelemetry.javaagent.instrumentation.netty.v4_1.server.HttpServerResponseTracingHandler",
+      "io.opentelemetry.javaagent.instrumentation.netty.v4_1.server.HttpServerTracingHandler"
+    };
+  }
```

Comment:
You basically just need to move `typeMatcher()` and `transformers()` into the new `TypeInstrumentation` implementation:

```suggestion
@AutoService(InstrumentationModule.class)
public final class ReactorHttpClientInstrumentationModule extends InstrumentationModule {

  public ReactorHttpClientInstrumentationModule() {
    super("reactor-httpclient");
  }

  @Override
  public String[] helperClassNames() {
    return new String[] {
      ReactorHttpClientInstrumentation.class.getName() + "$MapConnect",
      ReactorHttpClientInstrumentation.class.getName() + "$OnRequest",
      "io.opentelemetry.javaagent.instrumentation.netty.v4_1.AttributeKeys",
      "io.opentelemetry.javaagent.instrumentation.netty.v4_1.AttributeKeys$1",
      // these below a transitive dependencies of AttributeKeys from above
      "io.opentelemetry.javaagent.instrumentation.netty.v4_1.client.NettyHttpClientTracer",
      "io.opentelemetry.javaagent.instrumentation.netty.v4_1.client.NettyResponseInjectAdapter",
      "io.opentelemetry.javaagent.instrumentation.netty.v4_1.client.HttpClientRequestTracingHandler",
      "io.opentelemetry.javaagent.instrumentation.netty.v4_1.client.HttpClientResponseTracingHandler",
      "io.opentelemetry.javaagent.instrumentation.netty.v4_1.client.HttpClientTracingHandler",
      "io.opentelemetry.javaagent.instrumentation.netty.v4_1.server.NettyHttpServerTracer",
      "io.opentelemetry.javaagent.instrumentation.netty.v4_1.server.NettyRequestExtractAdapter",
      "io.opentelemetry.javaagent.instrumentation.netty.v4_1.server.HttpServerRequestTracingHandler",
      "io.opentelemetry.javaagent.instrumentation.netty.v4_1.server.HttpServerResponseTracingHandler",
      "io.opentelemetry.javaagent.instrumentation.netty.v4_1.server.HttpServerTracingHandler"
    };
  }
  
  @Override
  public List<TypeInstrumentation> typeInstrumentations() {
    return singletonList(new HttpClientInstrumentation());
  }
  
  private static final class HttpClientInstrumentation implements TypeInstrumentation {
    @Override
    public ElementMatcher<TypeDescription> typeMatcher() {
      return named("reactor.netty.http.client.HttpClient");
    }

    @Override
    public Map<? extends ElementMatcher<? super MethodDescription>, String> transformers() {
      return singletonMap(
          isStatic().and(named("create")),
          ReactorHttpClientInstrumentation.class.getName() + "$CreateAdvice");
    }
  }
```